File: src/api/MintyPeterson.Counter.Api/Models/Responses/EntryViewResponse.cs
// <copyright file="EntryViewResponse.cs" company="Tom Cook">
// Copyright (c) Tom Cook. All rights reserved.
// </copyright>

namespace MintyPeterson.Counter.Api.Models.Responses
{
  /// <summary>
  /// Provides a response model for <see cref="Controllers.EntryController.ViewAsync"/>.
  /// </summary>
  public class EntryViewResponse
  {
    /// <summary>
    /// Gets or sets the entry identifier.
    /// </summary>
    public Guid EntryId { get; set; }

    /// <summary>
    /// Gets or sets the entry date.
    /// </summary>
    /// <remarks>Only the date part is retained.</remarks>
    public DateTime EntryDate { get; set; }

    /// <summary>
    /// Gets or sets the entry.
    /// </summary>
    public decimal Entry { get; set; }

    /// <summary>
    /// Gets or sets the notes.
    /// </summary>
    public string? Notes { get; set; }
  }
}

File: src/flutter/lib/services/counter/responses/entry_view_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:counter/services/counter/counter_service.dart';

part 'entry_view_response.g.dart';

/// The response model from the [CounterService] entry view endpoint.
@JsonSerializable(createToJson: false)
class EntryViewResponse {

  final String entryId;
  final DateTime entryDate;
  final int entry;
  final String? notes;
  const EntryViewResponse({
    required this.entryId,
    required this.entryDate,
    required this.entry,
    required this.notes,
  });

  factory EntryViewResponse.fromJson(Map<String, dynamic> json) =>
    _$EntryViewResponseFromJson(json);
}